### PR TITLE
Issue 92 delete item UI tdd

### PR DESCRIPTION
**Pull request by studentgrad0419**
What is included

1. Fixed behaviour for delete wishlist item
2. Added Steps for delete wishlist item (seeing length of the table)
3. Re-organized how retrieve works
4. Defined "retrieveAndUpdateWishlist" so other methods can get an updated wishlist as part of the button press
5. Added delete functionality (on pressing delete on item, it will delete and update the page)
6. Added to features file: bdd for delete 

What could be done if time enables:
Reading and confirming a specific ID no longer is there.

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -291,10 +291,18 @@ def delete_addresses(wishlist_id, item_id):
...
+    # Find the specified Wishlist
     wishlist = Wishlist.find(wishlist_id)
+
+    # Find the specified WishlistItem
     if wishlist:
-        wishlist.delete()
+        wishlist_item = None
+        for item in wishlist.items:
+            if item.id == item_id:
+                wishlist_item = item
```
Comment (jay-goettelmann-nyu) [inline service/routes.py:302]: You could save a couple lines here if you wanted by removing line 299 and calling item.delete() in line 302. (Then also deleting line 305).

File: service/static/js/rest_api.js [hunk 2]
```diff
@@ -356,4 +357,30 @@ $(function () {
...
+        $.ajax({
+            method: "DELETE",
+            url: `/wishlists/${tokens[0]}/items/${tokens[1]}`,
+            contentType: "application/json",
+            data: '',
+        }).done(function(res, statusText, jqXHR){
+            // update visible list w/o changing success flash
+            if (jqXHR.status === 204) {
+                // Success - update visible list without changing success flash
+                retrieveAndUpdateWishlist(tokens[0], false);
```
Comment (jay-goettelmann-nyu) [inline service/static/js/rest_api.js:376]: I think for the retrieveAndUpdateWishlist we probably shouldn't deal with the flash message at all (except on retrieval failure).  One thing to be cautious of when creating a function that has asyncronous internal calls (i.e. $.ajax() ).  You can have some weird bugs arise.  Like it's possible that the flash_message in line 377 might fire before the ajax calls have completed inside of retrieveAndUpdateWishlist.  I think for our purposes this is fine, but just raising it as food for thought.

**Review by jay-goettelmann-nyu — APPROVED** [review]
Left some comments but I think we can merge this.

File: service/routes.py (hunk 1 above)
Comment (studentgrad0419) [reply service/routes.py:302]: gotcha, if it doesn't affect functionallity I think I'll leave it be for now.

File: service/static/js/rest_api.js (hunk 2 above)
Comment (studentgrad0419) [reply service/static/js/rest_api.js:376]: Gotcha, I was considering this as a chain. I'm not too familiar with JS, so I'll look into how I can fix that.

File: features/steps/wishlist_item_ui_steps.py [hunk 3]
```diff
@@ -80,3 +81,45 @@ def step_impl(context, text_string, element_name):
...
+        )
+        assert table_visible
+        rows_found = len(
+            context.driver.find_elements(By.CSS_SELECTOR, f"{table_selector} tbody tr")
+        )
+        assert int(rows_found) == expected_rows
+
+    except TimeoutException:
+        # Print additional information for debugging
+        actual_rows = len(
```
[the remark this answers is not in the document]
Comment (studentgrad0419) [reply features/steps/wishlist_item_ui_steps.py:107]: I'll look into seeing if I can remove it. It was helpful as a debugging step.

File: features/steps/wishlist_item_ui_steps.py [hunk 4]
```diff
@@ -80,3 +81,45 @@ def step_impl(context, text_string, element_name):
...
+def step_impl(context, expected_rows, table_id):
+    table_selector = "#" + table_id
+    expected_rows = int(expected_rows)
+    rows_found = -1
+
+    try:
+        # Wait for the table to be visible
+        table_visible = WebDriverWait(context.driver, context.wait_seconds).until(
+            expected_conditions.visibility_of_element_located(
+                (By.CSS_SELECTOR, table_selector)
```
[the remark this answers is not in the document]
Comment (studentgrad0419) [reply features/steps/wishlist_item_ui_steps.py:96]: I tried replacing with "by.ID" but ran into issues with a timeout error.
see here: 
image

**Review by jay-goettelmann-nyu — APPROVED** [review]
Nice job on the Promise approach!

**Pull request merged**